src/g_editor.c

```diff
@@ -4894,6 +5056,11 @@ void g_editor_setup(void)
 
     class_addmethod(canvas_class, (t_method)canvas_disconnect,
         gensym("disconnect"), A_FLOAT, A_FLOAT, A_FLOAT, A_FLOAT, A_NULL);
+#ifdef HAVE_KEYBOARDNAV
+/* --------------- to be used with the console ----------------------- */
+    class_addmethod(canvas_class, (t_method)canvas_connect_with_undo,
+        gensym("connect_with_undo"), A_FLOAT, A_FLOAT, A_FLOAT, A_FLOAT, A_NULL);
```

Comment:
i'm not sure whether an undoable canvas-message should be exposed to the user (even if you mask it).
sending messages to the canvas is low-level and in general not undoable. introducing it here, will only make people ask why they can undo when sending the message from the canvas-prompt but not when sending it directly.

Reply:
I did that so connections made with kbdnav through the canvas console could be undone normally.

Select an outlet and press <kbd>Ctrl</kbd>+<kbd>Shift</kbd>+<kbd>Down</kbd>. An entry field will be pre-filled with the relevant information and relevant part you should replace is already selected so you can just type.

![2020-01-20_23-19-34](https://user-images.githubusercontent.com/11790621/72770183-d83d6400-3bdb-11ea-9b63-c1d1358d3301.gif)

I think it works great for connection distant objects but i couldn't figure out a way of doing this without exposing the method in some way, since the GUI is sending the msg to the core.

What can be done?

Reply:
Resolving now that we have the "Connect Objects" dedicated dialog